plugins/stylescombo/plugin.js

```diff
@@ -6,6 +6,30 @@
 ( function() {
 	'use strict';
 
+	function getCutOffIndex(elements, tagNames) {
+		// given an array of elements and an array of tag names to stop on,
+		// returns the earliest index of any of the tag names, or -1
+
+		var cutOffIndexes = [];
+		var elementNames = elements.map(function(element) {
```

Comment:
I pushed my attempt at refactoring based on this comment.  Let me know tomorrow if I misunderstood :/